inc/ticket.class.php

```diff
@@ -6422,6 +6425,82 @@ static function cronCreateInquest($task) {
    }
 
 
+   /**
+    * Cron for ticket's automatic purge
+    *
+    * @param Crontask $task Crontask object
+    *
+    * @return integer (0 : nothing done - 1 : done)
+   **/
+   static function cronPurgeTicket(Crontask $task) {
+      global $DB;
+
+      $ticket = new self();
+
+      //search entities
+      $tot = 0;
+
+      $entities = $DB->request(
+         [
+            'SELECT' => 'id',
+            'FROM'   => Entity::getTable(),
+         ]
+      );
+
+      foreach ($entities as $entity) {
+         $delay  = Entity::getUsedConfig('autopurge_delay', $entity['id'], '', Entity::CONFIG_NEVER);
+         if ($delay >= 0) {
+            $criteria = [
+               'FROM'   => $ticket->getTable(),
+               'WHERE'  => [
+                  'entities_id'  => $entity['id'],
+                  'status'       => $ticket->getClosedStatusArray(),
+               ]
+            ];
+
+            if ($delay > 0) {
+               $calendars_id = Entity::getUsedConfig('calendars_id', $entity['id']);
```

Comment:
I think this is not necessary, we want a purge of all tickets whose **closedate** (DB value) is older than X days.